ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
+            return error("Error occurred while resetting the password" + responseRecord.toString());
         }
     }
 
-    private isolated function get(@display {label: "Resource Path"} string path)
-                                  returns @display {label: "Result"} json|Error {
-        json|http:ClientError response = self.salesforceClient->get(path);
-        if response is json {
-            return response;
-        } else {
-            return checkAndSetErrorDetail(response);
-        }
-    }
-
-    # Gets an object record by ID.
+    # Reset user password
     #
-    # + sobject - SObject name 
-    # + id - SObject ID
-    # + fields - Fields to retrieve 
-    # + return - JSON result if successful or else `sfdc:Error`
-    #
-    # # Deprecated
-    # This function is deprecated as it returns a json response.
-    # Use the new and improved `getById(string sObject, string id)` function instead.
-    @display {label: "Get Record by ID"}
-    @deprecated
-    isolated remote function getRecordById(@display {label: "SObject Name"} string sobject,
-                                            @display {label: "SObject ID"} string id,
-                                            @display {label: "Fields to Retrieve"}
-                                           string... fields)
-                                            returns @display {label: "Result"} json|Error {
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sobject, id]);
-        if fields.length() > 0 {
-            path = path.concat(utils:appendQueryParams(fields));
-        }
-        json response = check self.get(path);
-        return response;
-    }
-
-    private isolated function getByIdUtil(@display {label: "SObject Name"} string sobject,
-                                            @display {label: "SObject ID"} string id,
-                                            @display {label: "Fields to Retrieve"}
-                                          string... fields)
-                                            returns @display {label: "Result"} json|Error {
-        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sobject, id]);
-        if fields.length() > 0 {
-            path = path.concat(utils:appendQueryParams(fields));
+    # + userId - User ID
+    # + return - `byte[]` if successful or else `error`
+    isolated remote function resetPassword(string userId) returns byte[]|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, USER, userId, PASSWORD]);
+        http:Response response = check self.salesforceClient->delete(path);
+        if response.statusCode == 200 {
+            json payload = check response.getJsonPayload();
+            map<json> payloadMap = check payload.ensureType();
+            string password = check payloadMap.NewPassword;
+            return password.toBytes();
```

Comment:
Can we read the password as byte[] directly instead of string value?

Reply:
Updated in [update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/311/commits/cd6ae536defa51bd247b84742f1c480c2cf80ab1)